nncf/quantization/algorithms/channel_alignment/algorithm.py

```diff
@@ -0,0 +1,475 @@
+        patterns = self._get_target_patterns()
+        for subgraph in nncf_graph.find_matching_subgraphs(patterns):
+            if len(subgraph) == 2:
+                add_in = None
+                conv_in, conv_out = subgraph
+            else:
+                conv_in, add_in, conv_out = subgraph
+
+            if not self._check_producer_conv_node(conv_in):
+                continue
+
+            if not self._check_consumer_conv_node(conv_out):
+                continue
+
+            pairs.append((conv_in, add_in, conv_out))
+        return pairs
+
+    def _get_target_point_and_node_in(self, conv_in, add_in) -> Tuple[TargetPoint, NNCFNode]:
+        node_in = conv_in if add_in is None else add_in
+        input_port_id, _ = self._backend_entity.get_activation_port_ids_for_node(node_in)
+        return (
+            self._backend_entity.target_point(TargetType.POST_LAYER_OPERATION, node_in.node_name, input_port_id),
+            node_in,
+        )
+
+    def get_statistic_points(self, model: TModel) -> StatisticPointsContainer:
+        self._set_backend_entity(model)
+        self._original_nncf_graph = NNCFGraphFactory.create(model)
+
+        statistic_container = StatisticPointsContainer()
+        for conv_in, add_in, _ in self._get_node_pairs(self._original_nncf_graph):
+            target_point, node_in = self._get_target_point_and_node_in(conv_in, add_in)
+            channel_axis = conv_in.metatype.output_channel_axis
+            reduction_shape = list(range(len(self._original_nncf_graph.get_output_edges(node_in)[0].tensor_shape)))
+            reduction_shape.remove(channel_axis)
+
+            statistic_collector = self._backend_entity.get_statistic_collector(
+                tuple(reduction_shape), self._quantile, self.subset_size, self.inplace_statistics
+            )
+            statistic_container.add_statistic_point(
+                StatisticPoint(
+                    target_point=target_point,
+                    tensor_collector=statistic_collector,
+                    algorithm=ChannelAlignment,
+                )
+            )
+
+        return statistic_container
+
+
+class StatedTensor:
+    """
+    Tensor wrapper with additional method is_modified which is true if
+    given tensor was modified at least once after the initialization.
+    """
+
+    def __init__(self, value: np.ndarray):
+        """
+        :param value: Tensor to wrap.
+        """
+        self._value = value
+        self._mod_times = 0
+
+    @property
+    def val(self):
+        return self._value
+
+    @val.setter
+    def val(self, value):
+        if self._value is None and value is None:
+            return
+        self._mod_times += 1
+        self._value = value
+
+    def is_modified(self) -> bool:
+        """
+        :return: True if wrapped tensor was changed at least once after the
+            initialization else False.
+        """
+        return self._mod_times > 0
+
+
+class ConvParamsContainer:
```

Comment:
Do you really need this class? You use it only in single function

Reply:
What do you suggest? I need a structure to keep weights/bias/dims_descr for a conv node, what is your concerns?

Reply:
I don't want to create 5 similar variables for each convolution, it would be harder to read, test and maintain

Reply:
Why do you need to have a state of bias and weight? I didn't get this part

Reply:
I'm checking whether weights/biases were actually updated or not. If there was no changes, nobody assign any value to weight/bias, then update operation is not needed